### PR TITLE
Improve linkTo documentation

### PR DESCRIPTION
**Pull request by joscha**
The linkTo docs are incomplete, references https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19499

### CODE REVIEW

File: addons/links/README.md [hunk 1]
```diff
@@ -36,12 +36,14 @@ Have a look at the linkTo function:
 import { linkTo } from '@storybook/addon-links'
 
 linkTo('Toggle', 'off')
+linkTo(() => 'Toggle', () => 'off')
```
Comment (ndelangen) [inline addons/links/README.md]: I do not understand this example, what does this do?

**Comment by joscha** [comment]
It's the exact same as above - it was introduced in the PR mentioned in the
README, but never formally written down. Basically a callback-based link.

**Comment by ndelangen** [comment]
Interesting, TIL we have callback based linkTo 👍

**Review by ndelangen — APPROVED** [review]

File: addons/links/README.md (hunk 1 above)
Comment (UsulPro) [inline addons/links/README.md]: shouldn't it be?
```js
linkTo((event) => 'Toggle', (event) => 'off')
```

**Pull request merged**